ehrql/query_language.py

```diff
@@ -51,7 +51,7 @@ def __setattr__(self, name, value):
             )
         if name in self.variables:
             raise AttributeError(f"'{name}' is already set and cannot be reassigned")
-        if name == "variables":
+        if name in ("patient_id", "variables"):
             raise AttributeError("'variables' is not an allowed variable name")
```

Comment:
```suggestion
            raise AttributeError(f"'{name}' is not an allowed variable name")
```

Reply:
Ah, you beat me to it!

Reply:
:facepalm: Thanks Seb (and Iain). It is very, very hot.